### PR TITLE
[WIP] - Added support for Appcenter CI

### PR DESCRIPTION
**Pull request by RishabhTayal**
Fixes: #1056

### CODE REVIEW

File: lib/danger/ci_source/appcenter.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      @supported_request_sources ||= [
+        Danger::RequestSources::GitHub,
+        Danger::RequestSources::GitLab,
+        Danger::RequestSources::BitbucketServer,
+        Danger::RequestSources::BitbucketCloud
+      ]
+    end
+
+    def initialize(env)
+      self.pull_request_id = env["BITRISE_PULL_REQUEST"]
```
Comment (orta) [inline lib/danger/ci_source/appcenter.rb]: Bitrise?

Comment (RishabhTayal) [reply lib/danger/ci_source/appcenter.rb]: Still working on it.

Seems like Appcenter environment variables doesn't have any reference to pull request id. 🤷‍♂️

Do you know any way to get it?

Comment (RishabhTayal) [reply lib/danger/ci_source/appcenter.rb]: These are all the environment variables in Appcenter system

```
ISROOT
SYSTEM_TEAMFOUNDATIONCOLLECTIONURI
BUILD_SOURCEBRANCH
SYSTEM_TASKDEFINITIONSURI
AGENT_VERSION
SYSTEM_JOBATTEMPT
DANGER_GITHUB_API_TOKEN
BUILD_QUEUEDBY
SYSTEM_COLLECTIONURI
SYSTEM_HOSTTYPE
SYSTEM_JOBPARALLELISMTAG
BUILD_REPOSITORY_GIT_SUBMODULECHECKOUT
ANDROID_HOME
NVM_CD_FLAGS
MD_APPLE_SDK_ROOT
RUNTESTS
SHELL
MOBILECENTER_SOURCE_DIRECTORY
BUILD_STAGINGDIRECTORY
AGENT_MACHINENAME
TMPDIR
MODULENAME
APPCENTER_BRANCH
PR_CHECK_ID
SYSTEM_WORKFOLDER
COMMON_TESTRESULTSDIRECTORY
APPCENTER_ANDROID_MODULE
Apple_PubSub_Socket_Render
AGENT_JOBNAME
BUILDVARIANT
MSDEPLOY_HTTP_USER_AGENT
JAVA_HOME_8_X64
XCODE_9_DEVELOPER_DIR
RCT_NO_LAUNCH_PACKAGER
AGENT_OSARCHITECTURE
BUILD_REQUESTEDFOREMAIL
NUNIT_BASE_PATH
AGENT_ACCEPTTEEEULA
NUNIT3_PATH
LC_ALL
SYSTEM_CULTURE
APPCENTER_ANDROID_VARIANT
SYSTEM_DEFINITIONNAME
INPUT_ARGS
JAVA_HOME_11_X64
BUILD_REPOSITORY_PROVIDER
BUILD_REPOSITORY_CLEAN
BUILD_SOURCEBRANCHNAME
AGENT_TEMPDIRECTORY
USER
NVM_DIR
SYSTEM_JOBIDENTIFIER
XCODE_10_DEVELOPER_DIR
SYSTEM_TEAMFOUNDATIONSERVERURI
TF_BUILD
SYSTEM_TASKDISPLAYNAME
AZURE_HTTP_USER_AGENT
BUILD_QUEUEDBYID
APPCENTER_OUTPUT_DIRECTORY
SSH_AUTH_SOCK
VSTS_PROCESS_LOOKUP_ID
AGENT_ROOTDIRECTORY
SYSTEM_TEAMPROJECTID
BUILDSCRIPTS_TOOLSET
__CF_USER_TEXT_ENCODING
SYSTEM_TEAMPROJECT
AGENT_HOMEDIRECTORY
MOBILECENTER_BUILD_ID
AGENT_TOOLSDIRECTORY
BUILD_REPOSITORY_ID
BUILD_REPOSITORY_LOCALPATH
SYSTEM_JOBDISPLAYNAME
agent.jobstatus
AGENT_BUILDDIRECTORY
BUILD_REASON
SYSTEM_PIPELINESTARTTIME
SYSTEM
AGENT_OS
RUNLINT
BUILD_SOURCESDIRECTORY
PATH
SYSTEM_DEBUG
SYSTEM_ISSCHEDULED
MOBILECENTER_OUTPUT_DIRECTORY
_
MOBILECENTER_BRANCH
MOBILECENTER_ANDROID_MODULE
SYSTEM_PULLREQUEST_ISFORK
BUILD_BUILDURI
PWD
CONDA
SYSTEM_DEFINITIONID
JAVA_HOME
JAVA_HOME_7_X64
MOBILECENTER_TRIGGER
LANG
APPCENTER_BUILD_ID
SYSTEM_TASKINSTANCENAME
SYSTEM_PHASEDISPLAYNAME
BUILD_REPOSITORY_NAME
SYSTEM_SERVERTYPE
XPC_FLAGS
BUILD_REPOSITORY_URI
BUILD_DEFINITIONNAME
AGENT_WORKFOLDER
JAVA_HOME_9_X64
BUILD_REQUESTEDFOR
SYSTEM_JOBNAME
XPC_SERVICE_NAME
SYSTEM_TIMELINEID
SYSTEM_ARTIFACTSDIRECTORY
AGENT_ID
SHLVL
HOME
JAVA_HOME_10_X64
XCODE_8_DEVELOPER_DIR
APPCENTER_TRIGGER
AGENT_RETAINDEFAULTENCODING
BUILD_BINARIESDIRECTORY
BUILD_REQUESTEDFORID
BUILD_ARTIFACTSTAGINGDIRECTORY
BUILD_BUILDID
AUTOMATICSIGNING
MOBILECENTER_ANDROID_VARIANT
SYSTEM_TASKINSTANCEID
SONOMA_TAGS
LOGNAME
BUILD_SOURCEVERSION
LC_CTYPE
USER-DEFINED_DANGER_GITHUB_API_TOKEN
PKG_CONFIG_PATH
SYSTEM_DEFAULTWORKINGDIRECTORY
SYSTEM_JOBID
ANDROID_NDK_HOME
AGENT_NAME
BUILD_DEFINITIONVERSION
SYSTEM_PLANID
SYSTEM_PHASEID
TASK_DISPLAYNAME
SYSTEM_COLLECTIONID
AGENT_JOBSTATUS
ENDPOINT_URL_SYSTEMVSSCONNECTION
APPCENTER_SOURCE_DIRECTORY
SYSTEM_PHASENAME
BUILD_BUILDNUMBER
SECRET_KEYSTOREENCODED
BUILD_CONTAINERID
```

Comment (orta) [reply lib/danger/ci_source/appcenter.rb]: `PR_CHECK_ID` might be something you could use an API to work with maybe?

Comment (RishabhTayal) [reply lib/danger/ci_source/appcenter.rb]: `PR_CHECK_ID=44920015` while pr url is `https://github.com/username/repo/pull/289`. It doesn't match.

Comment (orta) [reply lib/danger/ci_source/appcenter.rb]: A check is PR related metadata, so you can probably go back from that to the PR: https://developer.github.com/v3/checks/suites/

Comment (RishabhTayal) [reply lib/danger/ci_source/appcenter.rb]: so we would need to call [github's api](https://developer.github.com/v3/checks/suites/#get-a-single-check-suite) to get the PR info?

Comment (orta) [reply lib/danger/ci_source/appcenter.rb]: I think so, yep, which means this can only support GitHub

Comment (RishabhTayal) [reply lib/danger/ci_source/appcenter.rb]: can you point to existing code base (if we have any) which calls github apis?

**Comment by orta** [comment]
This looks pretty close, you want to continue?

**Comment by RishabhTayal** [comment]
Not able to get the PR id from appcenter environment variables. Contacted their team about it. They don't expose that variable.

**Comment by RishabhTayal** [comment]
Would be great if @orta you could help me with getting the PR id using Github api. Not able to figure out how to do that.

**Comment by orta** [comment]
I gave you an example of how another CI source does it - see codeship.rb 👍

**Comment by cojoj** [comment]
I've conctacted App Center's support and they asked me to create a **feature request**. With this I guess it'd be way easier to create Danger's integration to the App Center. Would be greate to bump this request: https://github.com/Microsoft/appcenter/issues/79

**Comment by orta** [comment]
Fixed in #1086

**Comment by cojoj** [comment]
App Center integration has been added in #1086. I guess we can close this one now, right? 😉

**Pull request closed without merging**